qiskit/circuit/quantumcircuit.py

```diff
@@ -243,23 +243,90 @@ def has_register(self, register):
+                      reverse the order of gates.', DeprecationWarning)
+        return circuit.reverse()
+
+    def reverse(self):
+        """Reverse the circuit by reversing the order of instructions.
 
-        This is done by recursively mirroring all instructions.
-        It does not invert any gate.
+        This is done by recursively reversing all instructions.
+        It does not invert (adjoint) any gate.
 
         Returns:
-            QuantumCircuit: the mirrored circuit
+            QuantumCircuit: the reversed circuit.
+
+        Examples:
+
+            input:
+                 ┌───┐
+            q_0: ┤ H ├─────■──────
+                 └───┘┌────┴─────┐
+            q_1: ─────┤ RX(1.57) ├
+                      └──────────┘
+
+            output:
+                             ┌───┐
+            q_0: ─────■──────┤ H ├
+                 ┌────┴─────┐└───┘
+            q_1: ┤ RX(1.57) ├─────
+                 └──────────┘
         """
         reverse_circ = QuantumCircuit(*self.qregs, *self.cregs,
-                                      name=self.name + '_mirror')
+                                      name=self.name + '_reverse')
 
         for inst, qargs, cargs in reversed(self.data):
-            reverse_circ._append(inst.mirror(), qargs, cargs)
+            reverse_circ._append(inst.reverse(), qargs, cargs)
         return reverse_circ
 
+    def reverse_bits(self):
+        """Return a circuit with the opposite order of wires.
+
+        The circuit is "vertically" flipped. If a circuit is
+        defined over multiple registers, the resulting circuit will have
+        the same registers but with their order flipped.
+
+        This method is useful for converting a circuit written in little-endian
+        convention to the big-endian equivalent, and vice versa.
+
+        Returns:
+            QuantumCircuit: the circuit with reversed bit order.
+
+        Examples:
+
+            input:
+                 ┌───┐
+            q_0: ┤ H ├─────■──────
+                 └───┘┌────┴─────┐
+            q_1: ─────┤ RX(1.57) ├
+                      └──────────┘
+
+            output:
+                      ┌──────────┐
+            q_0: ─────┤ RX(1.57) ├
+                 ┌───┐└────┬─────┘
+            q_1: ┤ H ├─────■──────
+                 └───┘
+        """
+        circ = QuantumCircuit(*reversed(self.qregs), *reversed(self.cregs),
+                              name=self.name + '_r')
+        num_qubits = self.num_qubits
+        num_clbits = self.num_clbits
+        old_qubits = self.qubits
+        old_clbits = self.clbits
+        new_qubits = circ.qubits
+        new_clbits = circ.clbits
+
+        for inst, qargs, cargs in self.data:
+            new_qargs = [new_qubits[num_qubits - old_qubits.index(q) - 1] for q in qargs]
+            new_cargs = [new_clbits[num_clbits - old_clbits.index(c) - 1] for c in cargs]
+            circ._append(inst.reverse_bits(), new_qargs, new_cargs)
```

Comment:
I think there's a case where you could doubly reverse the bits here (`Circuit.reverse_bits` reverses the wires into the instruction, and `Instruction.reverse_bits` reverses the wiring inside the instruction.)

e.g.
```
>>> wide_cx = qk.QuantumCircuit(3)
>>> wide_cx.cx(0,1)
>>> wide_cx.cx(1,2)
>>> print(wide_cx.reverse_bits())

          ┌───┐
q_0: ─────┤ X ├
     ┌───┐└─┬─┘
q_1: ┤ X ├──■──
     └─┬─┘
q_2: ──■───────

>>> wide_cxg = wide_cx.to_gate()
>>> cx_box = qk.QuantumCircuit(3)
>>> cx_box.append(wide_cgx, [0,1,2])
>>> print(cx_box.decompose())

q_0: ──■───────
     ┌─┴─┐
q_1: ┤ X ├──■──
     └───┘┌─┴─┐
q_2: ─────┤ X ├
          └───┘
>>> print(cx_box.reverse_bits().decompose())


q_0: ──■───────
     ┌─┴─┐
q_1: ┤ X ├──■──
     └───┘┌─┴─┐
q_2: ─────┤ X ├
          └───┘
```

Reply:
Thanks for the catch. I updated and added tests for boxed circuits.